src/network/networkd-json.c

```diff
@@ -1050,6 +1051,79 @@ static int dhcp6_client_append_json(Link *link, JsonVariant **v) {
         return json_append_one(v, "DHCPv6Client", w);
 }
 
+static int dhcp4_client_pd_append_json(Link *link, JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *addresses = NULL, *array = NULL;
+        uint8_t ipv4masklen;
+        uint8_t sixrd_prefixlen;
+        struct in6_addr sixrd_prefix;
+        const struct in_addr *br_addresses;
+        size_t n_br_addresses = 0;
+        size_t n = 0;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp_lease || !dhcp4_lease_has_pd_prefix(link->dhcp_lease))
```

Comment:
Maybe, better to also check `dhcp_use_6rd` flag ?

Reply:
Done.